### PR TITLE
chore: Build once, run everywhere

### LINKED ISSUE #8101: Move Pushing client build to S3 out of webpack & into syncToCDN
Today, we push the client build to our CDN as part of our webpack build.
In a future where we build once and deploy to multiple PPMIs, we'll need to push the client build to multiple CDNs.
That requires decoupling the push from the build. 

We already do something similar with meetingTemplate images in #8071.  We can build off that.
We could special case our own cloud build, but it'd probably be cleaner to not have special cases & call syncToCDN for every predeployment we run. 

AC
- [x] chat with @rafaelromcar-parabol to figure out how best to do this (are client assets kept in the docker image)
- [x] syncToCDN does what the webpack S3Plugin does today: It pushes the entire build directory to the CDN listed in the .env
- [x] S3Plugin is removed from webpack & the app
- [x] favicon uses the runtime CDN, it is not hardcoded
- [x] initial JS/CSS in index.html point to the CDN
- [x] webpack's `publicPath` is not determined at build time. it is determined by the server at runtime using the webpack var or 'auto'
- [x] sourcemaps are available
- [x] local filestore finds the build files at /static
- [x] cdn filesstore finds the build files at /build/vX.Y.Z
- [ ] verify `jiraPlaceholderBuffer` is correct 
- [ ] pushing sourcemaps to sentry should occur in pushToCDN, not during the circleCI script
- [ ] serveStatic should only serve client assets
- [ ] only 1 reference to action.parabol.co being hardcoded
- [ ] no references to action-files.parabol.co hardcoded
- [ ] remove protocol relative uses of the CDN_BASE_URL
- [x] css link needs crossorigin in initial index.html
- [x] manifest to build dir
- [x] empty the build dir before writing to it
- [ ] dynamic fetches, like extra CSS, have a dynamic publicPath

Estimate: 8 hours

### PR DESCRIPTION
**Pull request by mattkrick**
# Description

fix #8101

In the near future, every PPMI will use the same docker image. The only difference between deploys will be environment variables. That means we need to remove ENV vars from the build step.

The last remaining env var that exist in the build step is CDN_BASE_URL. 

We can achieve this in 1 of 2 ways:
-  create a predeploy step that rewrites our client assets with a different CDN that each PPMI can call
- force every instance to use the same `publicPath` and if a client wants to use a CDN, they may do so behind their own reverse proxy.

I built both, & prefer the 2nd option, but that involves transforming requests from /static to a CDN. cloudflare only permits that at the enterprise tier. If we instead used a redirect, then all assets would require 2 round trips. if we did it at our reverse proxy, we'd be handling a bunch of extra load. if we did it in a worker, we'd have to put our whole app behind that worker to transform the pathname. So, i went with the first option, which means we still have CORS issues.

By moving this var to preDeploy, that means the public path for initial assets (index.html, sw.js, manifest.json, favicon.ico) must get defined at preDeploy. Dynamic assets (ie webpack chunks) have their publicPath defined at runtime. To do this, we must share the publicPath with the client so it knows where to fetch the next chunks. 
 
## Demo

[If possible, please include a screenshot or gif/video, it'll make it easier for reviewers to understand the scope of the changes and how the change is supposed to work. If you're introducing something new or changing the existing patterns, please share a Loom and explain what decisions you've made and under what circumstances]

## Testing scenarios

- [ ] run `yarn dev`. everything works as expected (if you have a service worker already installed, remove it & hard refresh first: cmd + shift + R)
- [ ] with FILE_STORE_PROVIDER='s3' and `CDN_BASE_URL='//action-files.parabol.co/development'`, run `yarn build && node dist/preDeploy.js && yarn start`. the assets get pushed to our S3 & the app is available localy. NOTE: assets are no longer stored in a different directory for each version! This is necessary to cache unchanged assets like images across builds!
- [ ] To verify caching across versions, change the text of a single component & run `yarn build && node dist/preDeploy.js && yarn start`. Open the console & see that the service worker is reusing many of the old assets.
- [ ] with `FILE_STORE_PROVIDER='local'` and `CDN_BASE_URL='//action-files.parabol.co/development'` run `node dist/preDeploy.js && yarn start`. notice an error saying that CDN_BASE_URL must be empty. This is because we don't allow persisting assets to local while exposing them to a CDN
- [ ] with `FILE_STORE_PROVIDER='local'` and `CDN_BASE_URL=''` run `node dist/preDeploy.js && yarn start`. assets get pushed locally & the app works as expected

### CODE REVIEW

File: static/images/brand/mark-cropped-192.png
Comment (mattkrick) [inline static/images/brand/mark-cropped-192.png:1]: chrome kept giving me an error saying our pics were off by 1 pixel so i fixed that here

**Review by igorlesnenko — CHANGES_REQUESTED** [review]
I see few places where URLs are hardcoded. Should we change them or remove if it is legacy?

https://github.com/ParabolInc/parabol/blob/b656e7d44061aabd78abadfb05e79d61de9bd181/packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/LogoFooter.tsx#L58

https://github.com/ParabolInc/parabol/blob/6c88e71f2bb11d4966c3b23c7a49144e9eb69782/.circleci/config.yml#L306-L314

https://github.com/ParabolInc/parabol/blob/69cad570a630f877f2f01aea79cd7fdd1431fb39/packages/client/types/constEnums.ts#L125-L126

https://github.com/ParabolInc/parabol/blob/f3361ab138d76ecbd2fa233e518790226414709e/packages/client/hooks/useMockPeekers.ts

File: packages/server/utils/StaticServer.ts [hunk 1]
```diff
@@ -2,7 +2,6 @@ import fs from 'fs'
 import mime from 'mime-types'
 import path from 'path'
 import {brotliCompressSync} from 'zlib'
-import PROD from '../PROD'
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply packages/server/utils/StaticServer.ts:5]: yes! `PROD` is from back when we didn't use webpack to compile the server. I wanted to do it in this PR, but it's already size L so i decided to do it in the next. Issue here https://github.com/ParabolInc/parabol/issues/8627

File: scripts/toolboxSrc/preDeploy.ts [hunk 2]
```diff
@@ -23,18 +23,19 @@ const storePersistedQueries = async () => {
...
   const r = await getRethink()
   const res = await r.table('QueryMap').insert(records, {conflict: 'replace'}).run()
+  await new Promise((resolve) => setTimeout(resolve, 50))
   await r.getPoolMaster()?.drain()
 
   console.log(`🔗 QueryMap Persistence Complete: ${res.inserted} records added`)
 }
 
 const preDeploy = async () => {
-  const envPath = path.join(PROJECT_ROOT, '.env')
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply scripts/toolboxSrc/preDeploy.ts:32]: yes! we've had that issue open for awhile here: https://github.com/ParabolInc/parabol/issues/8202. I'll be prioritizing that after this PR to keep the size down

**Comment by mattkrick** [comment]
> I see few places where URLs are hardcoded. Should we change them or remove if it is legacy?

Yes! We'll be getting rid of those next. https://github.com/ParabolInc/parabol/issues/8549

File: scripts/toolboxSrc/preDeploy.ts [hunk 3]
```diff
@@ -23,18 +23,19 @@ const storePersistedQueries = async () => {
 
   const r = await getRethink()
   const res = await r.table('QueryMap').insert(records, {conflict: 'replace'}).run()
+  await new Promise((resolve) => setTimeout(resolve, 50))
```
[the remark this answers is not in the document]
Comment (mattkrick) [reply scripts/toolboxSrc/preDeploy.ts:27]: comment added!

**Review by igorlesnenko — APPROVED** [review]

**Pull request merged**